File: src/game/tick/tick.service.ts
import type { TickListener } from "./tick.types";

export class TickService {
  private listeners: TickListener[] = [];

  public onTick(listener: TickListener): void {
    this.listeners.push(listener);
  }

  public tick(delta = 1): void {
    for (const listener of this.listeners) {
      listener(delta);
    }
  }
}

File: Mythlog/Tick/TickService.cs
namespace Mythlog.Tick;

public class TickService
{
    private readonly List<TickListener> _listeners = new();

    public void OnTick(TickListener listener)
    {
        _listeners.Add(listener);
    }

    public void Tick(float delta = 1f)
    {
        foreach (var listener in _listeners)
        {
            listener(delta);
        }
    }
}